src/data-source.ts

```diff
@@ -101,63 +109,33 @@ export class DataSource extends DataSourceWithBackend<RedisQuery, RedisDataSourc
      * Streaming enabled
      */
     return new Observable<DataQueryResponse>((subscriber) => {
-      /**
-       * This dataframe can have values constantly added, and will never exceed the given capacity
-       */
-      const frame = new CircularDataFrame({
-        append: 'tail',
-        capacity: refA?.streamingCapacity || 1000,
-      });
+      const { streamingDataType = StreamingDataType.TimeSeries } = refA;
 
       /**
-       * Set refId and Time field
+       * Apply frame formatted by streamingDataType
        */
-      frame.refId = refA.refId;
-      frame.addField({ name: 'time', type: FieldType.time });
+      let frame: TimeSeriesFormatter | DataFrameFormatter = new TimeSeriesFormatter(refA);
+      if (streamingDataType === StreamingDataType.DataFrame) {
+        frame = new DataFrameFormatter();
+      }
 
       /**
        * Interval
        */
       const intervalId = setInterval(async () => {
-        let values: { [index: string]: number } = { time: Date.now() };
-
         /**
-         * Run Query and filter time field out
+         * Run Query
          */
-        const fields = await super
-          .query(request)
-          .pipe(
-            switchMap$((response) => response.data),
-            map$((data: DataFrame) => data.fields.filter((field) => (field.name === 'time' ? false : true)))
-          )
-          .toPromise();
-
-        if (fields) {
-          /**
-           * Add fields to frame fields and return values
-           */
-          fields.map((field) =>
-            field.values.toArray().map((value) => {
-              if (frame.fields.length < fields.length + 1) {
-                frame.addField({
-                  name: field.name,
-                  type: field.type === FieldType.string && !isNaN(value) ? FieldType.number : field.type,
```

Comment:
If the string has numbers, a field should be number to plot on chart

Reply:
fixed